lab_2_tokenize_by_bpe/main.py

```diff
@@ -14,6 +14,15 @@ def prepare_word(
     :param end_of_word: a token that signifies the end of word
     :return: preprocessed word
     """
+    if (not isinstance(raw_word, str)
+            or not isinstance(start_of_word, str | None)
+            or not isinstance(end_of_word, str | None)):
+        return None
+    if start_of_word is None:
+        return tuple([sign for sign in raw_word] + [end_of_word])
```

Comment:
are you sure you need the for. And why do you think, if `start_of_word` is `None` that `end_of_word` is not?

Reply:
fixed